Fix detach of grid when it contains frozen columns

Artur- opened the pull request:
Depends on #8788

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/vaadin/framework/8803)
<!-- Reviewable:end -->

Artur- committed:
Fix detach of grid when it contains frozen columns

Fixes #8748

pleku commented:
Reviewed 7 of 7 files at r1.
Review status: all files reviewed at latest revision, 2 unresolved discussions.

---

*[server/src/main/java/com/vaadin/ui/Grid.java, line 2351 at r1](https://reviewable.io:443/reviews/vaadin/framework/8803#-KeoLZwga09Mn2kjR3tI:-KeoLZwga09Mn2kjR3tJ:bwphrlt) ([raw file](https://github.com/vaadin/framework/blob/8758f160959f76fbec7ee1ed62d72d01b150ecda/server/src/main/java/com/vaadin/ui/Grid.java#L2351)):*
> ```Java
>             getState(true).columnOrder.remove(columnId);
> 
>             assert displayIndex != -1;
> ```

message ?

---

*[server/src/test/java/com/vaadin/tests/server/component/grid/GridTest.java, line 451 at r1](https://reviewable.io:443/reviews/vaadin/framework/8803#-KeoLuajnJbuluYI70Il:-KeoLuajnJbuluYI70Im:b-owfhmd) ([raw file](https://github.com/vaadin/framework/blob/8758f160959f76fbec7ee1ed62d72d01b150ecda/server/src/test/java/com/vaadin/tests/server/component/grid/GridTest.java#L451)):*
> ```Java
> 
>         grid.removeColumn(grid.getColumns().get(0));
>         assertEquals("Frozen column count should update when removing last row",
> ```

Removing last ? Removing row ?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/framework/8803)*
<!-- Sent from Reviewable.io -->

Artur- commented:
Review status: all files reviewed at latest revision, 2 unresolved discussions.

---

*[server/src/main/java/com/vaadin/ui/Grid.java, line 2351 at r1](https://reviewable.io:443/reviews/vaadin/framework/8803#-KeoLZwga09Mn2kjR3tI:-Kep0OgVDX63We0pl6ZG:bu481go) ([raw file](https://github.com/vaadin/framework/blob/8758f160959f76fbec7ee1ed62d72d01b150ecda/server/src/main/java/com/vaadin/ui/Grid.java#L2351)):*
<details><summary><i>Previously, pleku (Pekka Hyvönen) wrote…</i></summary><blockquote>

message ?
</blockquote></details>

Fixed in #8788 

---

*[server/src/test/java/com/vaadin/tests/server/component/grid/GridTest.java, line 451 at r1](https://reviewable.io:443/reviews/vaadin/framework/8803#-KeoLuajnJbuluYI70Il:-Kep0VeCw-p_WS3IbUsA:bu481go) ([raw file](https://github.com/vaadin/framework/blob/8758f160959f76fbec7ee1ed62d72d01b150ecda/server/src/test/java/com/vaadin/tests/server/component/grid/GridTest.java#L451)):*
<details><summary><i>Previously, pleku (Pekka Hyvönen) wrote…</i></summary><blockquote>

Removing last ? Removing row ?
</blockquote></details>

Fixed in #8788 

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/framework/8803)*
<!-- Sent from Reviewable.io -->

pleku commented:
Congrats, you get to do a manual rebase

---

Review status: all files reviewed at latest revision, all discussions resolved, some commit checks failed.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/framework/8803#-:-Kesl8ohNgSerqmLiSEQ:bs56hk)*
<!-- Sent from Reviewable.io -->
